tests/InstrEngineTests/Baselines/AddExceptionHandler_ReturnMultiVoidMethodTest.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0"?>
+<InstrumentedMethod>
+    <Name>ReturnMultiVoidMethodTest</Name>
+    <FullName>AddExceptionHandlerTests.Program.ReturnMultiVoidMethodTest</FullName>
+    <ClassID Volatile="True">0xd4a03998</ClassID>
+    <FunctionID>0x00000001</FunctionID>
+    <MethodToken>0x06000008</MethodToken>
+    <GenericParameterCount>0x00000000</GenericParameterCount>
+    <IsStatic>1</IsStatic>
+    <IsPublic>0</IsPublic>
+    <IsPrivate>1</IsPrivate>
+    <IsPropGetter>0</IsPropGetter>
+    <IsPropSetter>0</IsPropSetter>
+    <IsFinalizer>0</IsFinalizer>
+    <IsConstructor>0</IsConstructor>
+    <IsStaticConstructor>0</IsStaticConstructor>
+    <DeclaringTypeToken>0x02000002</DeclaringTypeToken>
+    <RetValueCorElementType>ELEMENT_TYPE_VOID</RetValueCorElementType>
+    <RetTypeToken>0x00000000</RetTypeToken>
+    <CorSignature>0x00 0x00 0x01 </CorSignature>
+    <CorMethodAttr>0x00000091</CorMethodAttr>
+    <CodeRVA>0x0000223e</CodeRVA>
+    <MethodImpl>0x00000000</MethodImpl>
+    <LocalSignature>0x07 0x00 </LocalSignature>
+    <MaxStack>0x00000002</MaxStack>
+    <Instructions>
+        <![CDATA[
+IL_0000 ldsfld 0x4000001
+IL_0005 ldc.i4.m1
+IL_0006 bne.un IL_0010
+IL_000b br IL_001b
```

Comment:
Something is confusing me about this test. When I compile this on my box, the IL I get is the following:

.method private hidebysig static void  ReturnMultiVoidMethodTest() cil managed
{
  // Code size       16 (0x10)
  .maxstack  8
  IL_0000:  ldsfld     int32 AddExceptionHandlerTests.Program::'error'
  IL_0005:  ldc.i4.m1
  IL_0006:  bne.un.s   IL_0009
  IL_0008:  ret
  IL_0009:  ldc.i4.0
  IL_000a:  stsfld     int32 AddExceptionHandlerTests.Program::'error'
  IL_000f:  ret
} // end of method Program::ReturnMultiVoidMethodTest

You can see that in the original set of instructions, this branch instruction is a return. I can't find anywhere in your instrumentation where you replace the return instruction. It is unclear to me how you are correctly handling multiple returns. Am I missing something?

Reply:
I don't handle multi return, I add the <MakeSingleRet /> node to the test script. I do this because I plan to convert any methods I instrument to single return, but I think it's interesting to test the results of an conversion.

See the test script here: https://github.com/microsoft/CLRInstrumentationEngine/pull/353/files#diff-11750276abb662ba0b270967e0279ef2b4056ef6baec3ddfeff8a6fd13fd6382R7

Reply:
OK, that makes sense. Can you swap the ordering in the XML of the MakeSingleRet, and add a comment to make it explicit? I'll add comments to the files that I'm thinking about as well.

---
In reply to: [566315167](https://github.com/microsoft/CLRInstrumentationEngine/pull/353#discussion_r566315167) [](ancestors = 566315167)

Reply:
Okay, fixed we this commit: https://github.com/robertpi/CLRInstrumentationEngine/commit/d01b06ef411a69720c80d5d59f4f997120d2c092

I didn't add a comment as I changed all the test and file names, so I think that makes it clear.